pkg/cmd/repo/clone/clone_test.go

```diff
@@ -331,3 +342,56 @@ func Test_RepoClone_withoutUsername(t *testing.T) {
 	assert.Equal(t, "", output.String())
 	assert.Equal(t, "", output.Stderr())
 }
+
+func TestSimplifyURL(t *testing.T) {
+	tests := []struct {
+		name        string
+		raw         string
+		expectedRaw string
+	}{
+		{
+			name:        "empty",
+			raw:         "",
+			expectedRaw: "",
+		},
+		{
+			name:        "no change, no path",
+			raw:         "https://github.com",
+			expectedRaw: "https://github.com",
+		},
+		{
+			name:        "no change, single part path",
+			raw:         "https://github.com/owner",
+			expectedRaw: "https://github.com/owner",
+		},
+		{
+			name:        "no change, two-part path",
+			raw:         "https://github.com/owner/repo",
+			expectedRaw: "https://github.com/owner/repo",
+		},
+		{
+			name:        "no change, three-part path",
+			raw:         "https://github.com/owner/repo/pulls",
+			expectedRaw: "https://github.com/owner/repo",
+		},
+		{
+			name:        "no change, two-part path, with query, with fragment",
+			raw:         "https://github.com/owner/repo?key=value#fragment",
```

Comment:
**Question**

Is there a case in which we would expect to be stripping query params in the real world? I ask because the original issue was about cloning from in the file tree, which should just be extra path entries.

Reply:
I was just a bit extra defensive on this and wanted to cover cases where one might copy a PR link (e.g., a link to a comment) or a deep-link to a line in a file. In these cases, there'll be a fragment at the end of the URL. I can't imagine what kind of links could come with a query parameter, but again decided to strip them away to be safe.

I agree this is so niche and could never happen in reality. If you think there's no value in that, I can update the PR.

Reply:
It's probably more complicated to not do it. We'd have to error out if someone provided a URL with a query string. I think the main issue is that when I read code I **really** want to know what user flow it is supporting. I think the best thing to do would be to expand on your `simplifyURL` comment to include when this is important e.g. if the user copies and pastes from the file tree browser or from a comment URL.

Reply:
I'll just update the godoc.

Reply:
Done. Please have a look.

Reply:
Yeh that's great thanks.